xarray/core/variable.py

```diff
@@ -410,7 +410,11 @@ def copy(self, deep=True):
         If `deep=True`, the data array is loaded into memory and copied onto
         the new object. Dimensions, attributes and encodings are always copied.
         """
-        if deep and not isinstance(self.data, dask_array_type):
+        if isinstance(self._data, PandasIndexAdapter):
+            # there is no need to copy the index values here even if deep=True
+            # since pandas.Index objects are immutable
+            data = PandasIndexAdapter(self) if deep else self._data
```

Comment:
Why do we need these two different code paths? Why not always return `self._data` in this case?